bin/portal-bridge/src/types/range.rs

```diff
@@ -0,0 +1,95 @@
+use std::collections::HashMap;
+
+/// Converts a block range into a mapping of epoch indexes to block ranges.
+///
+/// An epoch is defined as a range of 8192 blocks, starting at 0.
+/// The resulting HashMap contains:
+/// - For epochs that are completely covered by the block range: None
+/// - For epochs that are partially covered: Some((start_block, end_block)) where start_block and
+///   end_block represent the actual block numbers that overlap with the epoch
+///
+/// # Returns
+/// HashMap mapping epoch indexes to either None or Some((start_block, end_block))
+pub fn block_range_to_epochs(start_block: u64, end_block: u64) -> HashMap<u64, Option<(u64, u64)>> {
+    let mut epoch_map = HashMap::new();
+    const EPOCH_SIZE: u64 = 8192;
```

Comment:
```suggestion
```

Isn't this a pretty common constant couldn't we import this from somewhere else?